storybooknative/src/styled/index.js

```diff
@@ -1,10 +1,25 @@
 // @flow
 
 import * as React from "react";
+import { Platform } from "react-native";
 import styled from "styled-components";
+import omit from "lodash.omit";
 
 import type { Style as _Style } from "../types/Style";
 
+type Styles = _Style | ((props: Object) => _Style);
+
+const reduceStyles = styles =>
+  ["ios", "android", "web"].reduce((acc, type) => {
```

Comment:
Could you explain why deleting an object property is a better approach in this case?
It is possible to use the same approach as before with destructuring the styles object.

```
const reduceStyles = (styles: _Style): _Style => {
  let { ios, android, web, ...style } = { ...styles };
  if (ios && Platform.OS === "ios") {
    style = {...style, ...ios};
  }
  if (android && Platform.OS === 'android') {
     style = { ...style, ...android };
   }
  if (web && Platform.OS === 'web') {
     style = { ...style, ...android };
   }
   return style;
}; 
```
and in that case we do not need any loop, any library, etc.

Let me know your thoughts 😃

Reply:
Yes, this should work. It was my first implementation, but because I love `reduce` and this is much shorter, I prefer my solution. 
But I am OK with your suggestion, so I will rewrite it to your way.